src/test/java/uk/gov/companieshouse/registeredemailaddressapi/integration/EligibilityControllerEmailMockingIntegrationTest.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.companieshouse.registeredemailaddressapi.integration;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.web.servlet.MockMvc;
+
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.registeredemailaddressapi.integration.utils.Helper;
+import uk.gov.companieshouse.registeredemailaddressapi.service.CompanyProfileService;
+
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@TestPropertySource(properties = {"env.name = livesbox"})
+class EligibilityControllerEmailMockingIntegrationTest {
+
+    Helper helper = new Helper();
+
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    protected CompanyProfileService companyProfileService;
+
+
+    @Test
+    void EligibilityEndpointMockEmailResponseTest() throws Exception {
+
+        final String companyNumber = "12345ERR";
+        CompanyProfileApi companyProfileApi = helper.generateCompanyProfileApi(companyNumber);
+        when(companyProfileService.getCompanyProfile(companyNumber)).thenReturn(companyProfileApi);
```

Comment:
surprised you have not taken up the given().willReturn() style in BDD for new tests or may be you prefer consistency with existing tests